### PR TITLE
Cleaning start

### PR DESCRIPTION
**Pull request by hunoutl**
Refactoring the repo.
AVEC & STLIB lib are gone. They don't have there place here. 
**to do -> add requirements**
sofagym folder contains the sofagym lib. 
We can find examples scene in env (was envs before but rename to env to imitate the gym lib). 
**to do -> i think it's strange how it is done right now. To me it's should be a repo root.** 
**to do -> make corrections everywhere to be sync with sofa 22.06. Some calls are deprecated...**

### CODE REVIEW

File: sofagym/rpc_server.py [hunk 1]
```diff
@@ -525,7 +525,7 @@ def start_scene(config, nb_actions):
     # Run the first client
     def deferredStart():
         sdict = str(config)
-        subprocess.run([config['python_version'], path+"common/rpc_client.py", sdict, str(nb_actions), str(port_rpc)],
+        subprocess.run([config['python_version'], path+"sofagym/rpc_client.py", sdict, str(nb_actions), str(port_rpc)],
```
Comment (EtienneMenager) [inline sofagym/rpc_server.py:528]: suggestion for another PR: modifier config["python_version"] par sys.executable pour sortir la version de python utilisé de la config.

**Review by EtienneMenager — APPROVED** [review]

**Review by alxbilger — COMMENTED** [review]
Are AVEC & STLIB used anywhere in the project? In an example?
If yes, documentation should be updated to explain how to use AVEC and STLIB as an external dependency. Is it your plan when you say "to do -> add requirements"?

**Comment by hunoutl** [comment]
> Are AVEC & STLIB used anywhere in the project? In an example? If yes, documentation should be updated to explain how to use AVEC and STLIB as an external dependency. Is it your plan when you say "to do -> add requirements"?

exactly

**Comment by p-shg** [comment]
> > Are AVEC & STLIB used anywhere in the project? In an example? If yes, documentation should be updated to explain how to use AVEC and STLIB as an external dependency. Is it your plan when you say "to do -> add requirements"?
> 
> exactly

STLIB was added to the project a long time ago when the master STLIB was not up to date with SofaPython3, it is probably not needed anymore.
AVEC was used by @EtienneMenager for an experiment in the paper but I don't think we need to carry it here. Etienne what do you think?

**Comment by EtienneMenager** [comment]
> > > Are AVEC & STLIB used anywhere in the project? In an example? If yes, documentation should be updated to explain how to use AVEC and STLIB as an external dependency. Is it your plan when you say "to do -> add requirements"?
> > 
> > 
> > exactly
> 
> STLIB was added to the project a long time ago when the master STLIB was not up to date with SofaPython3, it is probably not needed anymore. AVEC was used by @EtienneMenager for an experiment in the paper but I don't think we need to carry it here. Etienne what do you think?

Yes, AVEC was just to show that we could use any kind of RL algorithms, even the ones that are coming from the actual research. For the library it is not very important I think...

**Comment by hunoutl** [comment]
I started the readme update.
For the examples folder, after reading gym's documentation and its way of register envs, I think it's better to keep the structure that I had already started to put in place. Namely sofagym.envs which contains the basic Sofa/Gym environments.
to do -> a documentation or a template to add easily a new env for sofagym.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,61 @@
...
+## Installation
+### Prerequisites
+
+#### 1) SofaGym need a [Sofa installation](https://www.sofa-framework.org/community/doc/getting-started/build/linux/) v22.06
+with some mandatory plugins :
+* [SofaPython3](https://github.com/sofa-framework/SofaPython3) (fetchable within sofa)
+* [BeamAdapter](https://github.com/sofa-framework/BeamAdapter) (fetchable within sofa)
+* [SPLIB](https://github.com/SofaDefrost/SPLIB)
+* [STLIB](https://github.com/SofaDefrost/STLIB)
+* [SoftRobots](https://github.com/SofaDefrost/SoftRobots)
```
Comment (alxbilger) [inline README.md:18]: Why BeamAdapter and SoftRobots are not optional?

File: README.md [hunk 3]
```diff
@@ -1,7 +1,61 @@
...
+
+and optional plugins (for some examples): 
+* [ModelOrderReduction](https://github.com/SofaDefrost/ModelOrderReduction)
+
+[comment]: <> (SoftRobots.Inverseand Cosserat)
```
Comment (alxbilger) [inline README.md:23]:
```suggestion
[comment]: <> (SoftRobots.Inverse and Cosserat)
```

File: test_env.py [hunk 4]
```diff
@@ -28,10 +28,15 @@
 sys.path.insert(0, os.getcwd()+"/..")
 
 __import__('sofagym')
-name = ['multigaitrobot-v0',
-        'gripper-v0', 'trunk-v0', 'trunkcup-v0', 'diamondrobot-v0',
-        'maze-v0', 'simple_maze-v0', 'concentrictuberobot-v0']
-num = 2
+name = ['multigaitrobot-v0',#[ERROR]   [RequiredPlugin(ModelOrderReduction)] Plugin not found: "ModelOrderReduction"
```
Comment (alxbilger) [inline test_env.py:31]: I guess the comments are the errors you get. But they must not be in the code. I suggest to remove the comments

**Review by alxbilger — CHANGES_REQUESTED** [review]
Typo in requierements.txt -> requirements.txt (to update in readme too)

File: README.md (hunk 2 above)
Comment (hunoutl) [reply README.md:18]: They are used in almost all of the envs examples. However, actually, they are not really mandatory. I will check that and correct accordingly.

**Pull request merged**